src/Entity/IdeasWorkshop/Idea.php

```diff
@@ -0,0 +1,234 @@
+     */
+    private $needs;
+
+    /**
+     * @ORM\ManyToOne(targetEntity="AppBundle\Entity\Adherent")
+     */
+    private $adherent;
+
+    /**
+     * @var \DateTime
+     *
+     * @ORM\Column(type="datetime", nullable=true)
+     */
+    private $publishedAt;
+
+    /**
+     * @var Committee
+     *
+     * @ORM\ManyToOne(targetEntity="AppBundle\Entity\Committee")
+     */
+    private $committee;
+
+    /**
+     * @Assert\Choice(
+     *     callback={"AppBundle\Entity\IdeasWorkshop\IdeaStatusEnum", "toArray"},
+     *     strict=true,
+     * )
+     *
+     * @ORM\Column(length=11, options={"default": IdeaStatusEnum::IN_PROGRESS})
+     */
+    private $status;
+
+    /**
+     * @ORM\OneToMany(targetEntity="Answer", mappedBy="idea")
+     */
+    private $answers;
+
+    public function __construct(
+        UuidInterface $uuid,
+        string $name,
+        Adherent $adherent,
+        Category $category,
+        Theme $theme,
+        ?Committee $committee,
+        ?\DateTime $publishedAt = null,
+        string $status = IdeaStatusEnum::IN_PROGRESS
+    ) {
+        $this->uuid = $uuid;
+        $this->setName($name);
+        $this->adherent = $adherent;
+        $this->category = $category;
+        $this->theme = $theme;
+        $this->committee = $committee;
+        $this->publishedAt = $publishedAt;
+        $this->status = $status;
+        $this->needs = new ArrayCollection();
+        $this->answers = new ArrayCollection();
+    }
+
+    public static function createUuid(string $name): UuidInterface
+    {
+        return Uuid::uuid5(Uuid::NAMESPACE_OID, $name);
+    }
+
+    public function getTheme(): Theme
+    {
+        return $this->theme;
+    }
+
+    public function setTheme(Theme $theme): void
+    {
+        $this->theme = $theme;
+    }
+
+    public function getCategory(): Category
+    {
+        return $this->category;
+    }
+
+    public function setCategory(Category $category): void
+    {
+        $this->category = $category;
+    }
+
+    public function getNeeds(): ArrayCollection
+    {
+        return $this->needs;
+    }
+
+    public function addNeed(Need $need): void
+    {
+        if (!$this->needs->contains($need)) {
+            $this->needs->add($need);
+        }
+    }
+
+    public function removeNeed(Need $need): void
+    {
+        $this->needs->removeElement($need);
+    }
+
+    public function getAdherent(): Adherent
+    {
+        return $this->adherent;
+    }
+
+    public function setAdherent(Adherent $adherent): void
+    {
+        $this->adherent = $adherent;
+    }
+
+    public function getPublishedAt(): \DateTime
+    {
+        return $this->publishedAt;
+    }
+
+    public function setPublishedAt(\DateTime $publishedAt): void
+    {
+        $this->publishedAt = $publishedAt;
+    }
+
+    public function getCommittee(): Committee
+    {
+        return $this->committee;
+    }
+
+    public function setCommittee(Committee $committee): void
```

Comment:
`?Committee $committee` as it can be null

Reply:
:heavy_check_mark: